servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6726,6 +6726,49 @@ public void putTenancy(ResourceContext ctx, String tenantDomain, String provider
         if (authzServiceTokenOperation) {
             setupTenantAdminPolicyInProvider(ctx, provSvcDomain, provSvcName, tenantDomain,
                     false, auditRef, caller);
+
+            // If listed as service provider, register dependency between the provider service and the tenant domain (unless dependency exist
+            // following creation of a previous resource group)
+
+            tenancyRegisterDomainDependency(ctx, tenantDomain, provider, true, auditRef, caller);
+        }
+    }
+
+    private void tenancyRegisterDomainDependency(ResourceContext ctx, String tenantDomain, String provider, boolean authzServiceTokenOperation, String auditRef, String caller) {
+        String serviceToRegister = provider;
+        try {
+            serviceToRegister = getAuthorizedProviderService((RsrcCtxWrapper) ctx, serviceToRegister, authzServiceTokenOperation, caller);
+            if (serviceProviderManager.isServiceProvider(serviceToRegister)) {
+
+                DomainList domainList = dbService.listDomainDependencies(serviceToRegister);
+                if (!domainList.getNames().contains(tenantDomain)) {
+                    dbService.putDomainDependency(ctx, tenantDomain, serviceToRegister, auditRef, caller);
+                }
+            }
+        } catch (ResourceException ex) {
+            if (ex.getCode() == ResourceException.UNAUTHORIZED) {
+                LOG.warn("Not registering domain dependency: " + ex.getMessage());
+                return;
+            }
+            throw ex;
+        }
+    }
+
+    private void tenancyDeregisterDomainDependency(ResourceContext ctx, String tenantDomain, String provSvcDomain, String provSvcName, boolean authzServiceTokenOperation, String auditRef, String caller) {
+        String serviceToDeregister = provSvcDomain + "." + provSvcName;
+        try {
+            serviceToDeregister = getAuthorizedProviderService((RsrcCtxWrapper) ctx, serviceToDeregister, authzServiceTokenOperation, caller);
```

Comment:
same comment as above. what is the use case where the serviceToDeregister is not the correct value?